app/scripts/modules/core/src/widgets/spinners/Spinner.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from 'react';
+
+const Logo = require('./logo.svg');
+
+export interface ISpinnerProps {
+  size?: 'nano' | 'small' | 'medium' | 'large' | 'page';
+  message?: string;
+  postnote?: string;
+}
+
+export class Spinner extends React.Component<ISpinnerProps, {}> {
+  public getBarRows = (count: number) => {
```

Comment:
If you want to keep this a separate function, we like to use real class functions and then the @autoBindMethods decorator to keep things clean and consistent. (This would apply to all the functions in this class, just search the repo for @autoBindMethods for examples)